# Libraries

In [1]:
import numpy as np # Numerical computation
import pandas as pd # Dataset manipulation

pd.set_option('display.max_columns', None) # Num of displayed columns

# Visualization tools
import matplotlib
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# Libraries version

print('Numpy ver:', np.__version__)
print('Pandas ver:', pd.__version__)
print('Matplotlib ver:', matplotlib.__version__)
print('Seaborn ver:', sns.__version__)

Numpy ver: 1.20.1
Pandas ver: 1.2.4
Matplotlib ver: 3.3.4
Seaborn ver: 0.11.1


# Import Dataset

In [3]:
df = pd.read_csv('../dataset/online_shoppers_intention.csv')

# Overview
df.sample(5)

Administrative  Administrative_Duration  Informational  \
10480               0                 0.000000              0   
10341               0                 0.000000              0   
9649                5               173.541667              5   
1544                2                18.000000              0   
10389               0                 0.000000              0   

       Informational_Duration  ProductRelated  ProductRelated_Duration  \
10480                    0.00              16               898.758333   
10341                    0.00               1                 0.000000   
9649                   109.75              38              1859.382937   
1544                     0.00              30               474.500000   
10389                    0.00               8                96.000000   

       BounceRates  ExitRates  PageValues  SpecialDay Month  OperatingSystems  \
10480     0.006250   0.014583         0.0         0.0   Nov                 1   
10341     0.200000   0.200000         0.0         0.0   Dec                 1   
9649      0.004545   0.009091         0.0         0.0   Dec                 3   
1544      0.000000   0.003226         0.0         0.0   Mar                 1   
10389     0.025000   0.050000         0.0         0.0   Nov                 2   

       Browser  Region  TrafficType        VisitorType  Weekend  Revenue  
10480        1       1            1  Returning_Visitor    False     True  
10341        1       4            1  Returning_Visitor     True    False  
9649         2       8           10  Returning_Visitor     True    False  
1544         1       3            2  Returning_Visitor     True    False  
10389        4       9           20  Returning_Visitor    False    False

# Exploratory Data Analysis

## Descriptive Statistics

In [4]:
# Dataset Info
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12330 entries, 0 to 12329
Data columns (total 18 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Administrative           12330 non-null  int64  
 1   Administrative_Duration  12330 non-null  float64
 2   Informational            12330 non-null  int64  
 3   Informational_Duration   12330 non-null  float64
 4   ProductRelated           12330 non-null  int64  
 5   ProductRelated_Duration  12330 non-null  float64
 6   BounceRates              12330 non-null  float64
 7   ExitRates                12330 non-null  float64
 8   PageValues               12330 non-null  float64
 9   SpecialDay               12330 non-null  float64
 10  Month                    12330 non-null  object 
 11  OperatingSystems         12330 non-null  int64  
 12  Browser                  12330 non-null  int64  
 13  Region                   12330 non-null  int64  
 14  TrafficType           

Dataset terdiri dari 12330 observasi dan 18 field. Setiap nama kolom sudah merepresentasikan isinya. Setiap field sudah memiliki tipe data yang sesuai dengan konten yang dimuatnya.

In [5]:
# Missing Values
df.isnull().sum()

Administrative             0
Administrative_Duration    0
Informational              0
Informational_Duration     0
ProductRelated             0
ProductRelated_Duration    0
BounceRates                0
ExitRates                  0
PageValues                 0
SpecialDay                 0
Month                      0
OperatingSystems           0
Browser                    0
Region                     0
TrafficType                0
VisitorType                0
Weekend                    0
Revenue                    0
dtype: int64

Tidak ada nilai kosong (missing value) di setiap field.

In [6]:
# Numeric Fields Descriptive Statistics
num_fields = [field for field in df.columns if df[field].dtypes != 'O' and 
              field not in ['OperatingSystems','Browser', 'Region', 'TrafficType']]

df[num_fields].describe()

Administrative  Administrative_Duration  Informational  \
count    12330.000000             12330.000000   12330.000000   
mean         2.315166                80.818611       0.503569   
std          3.321784               176.779107       1.270156   
min          0.000000                 0.000000       0.000000   
25%          0.000000                 0.000000       0.000000   
50%          1.000000                 7.500000       0.000000   
75%          4.000000                93.256250       0.000000   
max         27.000000              3398.750000      24.000000   

       Informational_Duration  ProductRelated  ProductRelated_Duration  \
count            12330.000000    12330.000000             12330.000000   
mean                34.472398       31.731468              1194.746220   
std                140.749294       44.475503              1913.669288   
min                  0.000000        0.000000                 0.000000   
25%                  0.000000        7.000000               184.137500   
50%                  0.000000       18.000000               598.936905   
75%                  0.000000       38.000000              1464.157214   
max               2549.375000      705.000000             63973.522230   

        BounceRates     ExitRates    PageValues    SpecialDay  
count  12330.000000  12330.000000  12330.000000  12330.000000  
mean       0.022191      0.043073      5.889258      0.061427  
std        0.048488      0.048597     18.568437      0.198917  
min        0.000000      0.000000      0.000000      0.000000  
25%        0.000000      0.014286      0.000000      0.000000  
50%        0.003112      0.025156      0.000000      0.000000  
75%        0.016813      0.050000      0.000000      0.000000  
max        0.200000      0.200000    361.763742      1.000000

In [7]:
# Encoded Fields Descriptive Statistics
encoded = [field for field in df.columns if df[field].dtypes != 'O' and 
                     field not in num_fields]

for field in encoded:
    print(field, ':', df[field].nunique(), 'unique values')
    print(df[field].value_counts())

OperatingSystems : 8 unique values
2    6601
1    2585
3    2555
4     478
8      79
6      19
7       7
5       6
Name: OperatingSystems, dtype: int64
Browser : 13 unique values
2     7961
1     2462
4      736
5      467
6      174
10     163
8      135
3      105
13      61
7       49
12      10
11       6
9        1
Name: Browser, dtype: int64
Region : 9 unique values
1    4780
3    2403
4    1182
2    1136
6     805
7     761
9     511
8     434
5     318
Name: Region, dtype: int64
TrafficType : 20 unique values
2     3913
1     2451
3     2052
4     1069
13     738
10     450
6      444
8      343
5      260
11     247
20     198
9       42
7       40
15      38
19      17
14      13
18      10
16       3
12       1
17       1
Name: TrafficType, dtype: int64


In [8]:
# Categorical Fields Descriptive Statistics
cat_fields = [field for field in df.columns if df[field].dtypes == 'O' or df[field].dtypes == 'bool']

df[cat_fields].describe()

Month        VisitorType Weekend Revenue
count   12330              12330   12330   12330
unique     10                  3       2       2
top       May  Returning_Visitor   False   False
freq     3364              10551    9462   10422

Summary Analisis Statistik Deskriptif

Fitur numerik

1. Semua fitur numerik *(Andiminstrative, Administrative_Duration, Informational, Informational_Duration, ProductRelated, ProductRelated_Duration, BounceRates, ExitRates, PageValues, dan SpecialDay)* memiliki nilai **mean dan median yang berbeda**. Kemungkinan fitur-fitur tersebut memiliki **outlier** atau **tidak menyebar normal (skewed)**.
2. Semua fitur yang disebutkan pada poin 1 juga memiliki nilai **maksimum** yang cukup **jauh dari nilai Q3** yang mengindikasikan bahwa fitur-fitur tersebut memiliki **outlier**.
3. Beberapa fitur (Informational, Informational_Duration, PageValues, SpecialDay) terlihat **didominasi oleh nilai 0** dan memiliki nilai **maksimum** yang cukup **besar**. Indikasi kuat bahwa fitur-fitur tersebut memiliki **outlier**.

Fitur Kategorik

1. Kolom target (Revenue) memiliki **2 unique value** namun **sebagian besar (sekitar 85%) bernilai False**. Bisa dipertimbangkan untuk penanganan **data imbalance**.
2. Beberapa fitur kategorik yaitu Browser, Region, dan TrafficType memiliki data yang **terlalu banyak unique value**. Beberapa value minor mungkin bisa dikategorikan sebagai **Other**.
3. Terdapat fitur-fitur yang **didominasi oleh satu unique value** saja seperti OperatingSystems, Browser, VisitorType, dan Weekend. Fitur-fitur tersebut mungkin bisa **tidak disertakan dalam model**.

## Univariate Analysis

## Multivariate Analysis

## Business Insight